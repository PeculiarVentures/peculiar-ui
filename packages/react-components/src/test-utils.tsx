import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
