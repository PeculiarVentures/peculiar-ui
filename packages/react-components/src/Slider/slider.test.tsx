import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Slider } from './index';

describe('<Slider />', () => {
  it('should render with default styles', () => {
    const {
      asFragment,
    } = render(<Slider />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const {
      asFragment,
    } = render(<Slider className="my-class-name" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
