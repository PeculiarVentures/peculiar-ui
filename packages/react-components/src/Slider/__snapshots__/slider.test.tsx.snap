// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Slider /> > should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 16px;
  width: 100%;
}

.emotion-1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  width: 100%;
  height: 100%;
  outline: none;
  background-color: var(--pv-color-primary-shade-1);
  border-radius: 16px;
  margin: 0;
  padding: 0;
  cursor: pointer;
  -webkit-transition: background-color 200ms;
  transition: background-color 200ms;
}

.emotion-1::-webkit-slider-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: 16px;
  width: 16px;
  background-color: var(--pv-color-primary-tint-5);
  border: 1px solid var(--pv-color-primary-shade-3);
  border-radius: 50%;
  cursor: inherit;
  -webkit-transition: background-color 200ms,border-color 200ms;
  transition: background-color 200ms,border-color 200ms;
}

.emotion-1::-moz-range-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: 16px;
  width: 16px;
  background-color: var(--pv-color-primary-tint-5);
  border: 1px solid var(--pv-color-primary-shade-3);
  border-radius: 50%;
  cursor: inherit;
  -webkit-transition: background-color 200ms,border-color 200ms;
  transition: background-color 200ms,border-color 200ms;
}

.emotion-1:not(:disabled):hover {
  background-color: var(--pv-color-primary-tint-1);
}

.emotion-1:not(:disabled):focus-visible {
  background-color: var(--pv-color-primary-tint-2);
}

.emotion-1:disabled {
  cursor: not-allowed;
  background-color: var(--pv-color-gray-4);
}

.emotion-1:disabled::-webkit-slider-thumb {
  background-color: var(--pv-color-gray-8);
  border-color: var(--pv-color-gray-8);
}

.emotion-1:disabled::-moz-range-thumb {
  background-color: var(--pv-color-gray-8);
  border-color: var(--pv-color-gray-8);
}

<div
    class="my-class-name emotion-0"
  >
    <input
      class="emotion-1"
      max="100"
      min="0"
      step="1"
      type="range"
      value=""
    />
  </div>
</DocumentFragment>
`;

exports[`<Slider /> > should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 16px;
  width: 100%;
}

.emotion-1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  width: 100%;
  height: 100%;
  outline: none;
  background-color: var(--pv-color-primary-shade-1);
  border-radius: 16px;
  margin: 0;
  padding: 0;
  cursor: pointer;
  -webkit-transition: background-color 200ms;
  transition: background-color 200ms;
}

.emotion-1::-webkit-slider-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: 16px;
  width: 16px;
  background-color: var(--pv-color-primary-tint-5);
  border: 1px solid var(--pv-color-primary-shade-3);
  border-radius: 50%;
  cursor: inherit;
  -webkit-transition: background-color 200ms,border-color 200ms;
  transition: background-color 200ms,border-color 200ms;
}

.emotion-1::-moz-range-thumb {
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  height: 16px;
  width: 16px;
  background-color: var(--pv-color-primary-tint-5);
  border: 1px solid var(--pv-color-primary-shade-3);
  border-radius: 50%;
  cursor: inherit;
  -webkit-transition: background-color 200ms,border-color 200ms;
  transition: background-color 200ms,border-color 200ms;
}

.emotion-1:not(:disabled):hover {
  background-color: var(--pv-color-primary-tint-1);
}

.emotion-1:not(:disabled):focus-visible {
  background-color: var(--pv-color-primary-tint-2);
}

.emotion-1:disabled {
  cursor: not-allowed;
  background-color: var(--pv-color-gray-4);
}

.emotion-1:disabled::-webkit-slider-thumb {
  background-color: var(--pv-color-gray-8);
  border-color: var(--pv-color-gray-8);
}

.emotion-1:disabled::-moz-range-thumb {
  background-color: var(--pv-color-gray-8);
  border-color: var(--pv-color-gray-8);
}

<div
    class="emotion-0"
  >
    <input
      class="emotion-1"
      max="100"
      min="0"
      step="1"
      type="range"
      value=""
    />
  </div>
</DocumentFragment>
`;
